components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/graph/AuthenticationGraphConfig.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model.graph;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import javax.xml.namespace.QName;
+
+/**
+ * Model for Authentication Steps graph.
+ * Graph has a start node, set of other connected nodes.
+ */
+public class AuthenticationGraphConfig implements Serializable {
+
+    private static final long serialVersionUID = 1610571097373920966L;
+    private static final Log log = LogFactory.getLog(AuthenticationGraphConfig.class);
+
+    /* User can incorrectly configure cyclic graph. This will eliminate the cyclic graph evaluation. */
+    private static final int MAX_RECURSION_COUNT = 32;
+
+    private Node startNode;
+    private Node endNode;
+    private String name;
+    private Map<String, Node> nodesMap;
+
+    /**
+     * Builds the graph with Axiom.
+     *
+     * @param graphOM OM element of the graph.
+     * @return the built graph. Will return null if supplied OMElement being null or it is not of the correct type.
+     */
+    public static AuthenticationGraphConfig build(OMElement graphOM) {
```

Comment:
"build" method just produce an instance of AuthenticationGraphConfig and does manipulate instance variable  so having static method is not a problem